balloon.py

```diff
@@ -910,6 +910,122 @@ def get_wind(RapData,altitude):
 
     return (ve,vn)
 
+
+
+def getPositions(vz, lat, lon, alt, airspeed, head, querytime):
+    #Inputs:
+    #vz: vertical speed (ascent rate)                                               type: float
+    #lat: current latitude of balloon (radians)                                     type: float
+    #lon: current longitude of balloon (radians)                                    type: float
+    #alt: current altitude (meters above sea level)                                 type: float
+    #airspeed: magnitude of airspeed                                                type: float
+    #head: direction of planar airspeed, degrees positive clockwise from north      type: float
+    #querytime: end of desired time window (seconds)                                type: float
+
+    #Assumptions:
+        # balloon ascends at a constant speed vz, and airspeed has no vertical component
+        # change in latitude is not large enough to affect distance-to-longitude conversion
+        # airspeed is constant in both magnitude and direction throughout the time window
+        # Earth's radius is constant (ground level invariance)
+        # change in height is not large enough to affect distance-to-angle conversion
+
+    #Outputs:
+    #positions: extrapolated positions. Dataframe of lat, lon, with datetime index
+
+    #This function extrapolates the state of the balloon a time ts into the future
+    def nextState(a, ts):
```

Comment:
Functions within functions are not good practice.  Move this function outside the other function and refactor.